netbox/utilities/filters.py

```diff
@@ -111,6 +117,152 @@ def __init__(self, *args, **kwargs):
+        MACAddressField: {
+            'filter_class': MultiValueMACAddressFilter
+        },
+    })
+
+    @classmethod
+    def get_filters(cls):
+        """
+        Override filter generation to support dynamic lookup expressions for certain filter types.
+
+        For specific filter types, new filters are created based on defined lookup expressions in
+        the form `<field_name>__<lookup_expr>`
+        """
+        # TODO: once 3.6 is the minimum required version of python, change this to a bare super() call
+        # We have to do it this way in py3.5 becuase of django_filters.FilterSet's use of a metaclass
+        filters = super(django_filters.FilterSet, cls).get_filters()
+
+        new_filters = {}
+        for existing_filter_name, existing_filter in filters.items():
+            # Loop over existing filters to extract metadata by which to create new filters
+
+            # If the filter makes use of a custom filter method or lookup expression skip it
+            # as we cannot sanely handle these cases in a generic mannor
+            if existing_filter.method is not None or existing_filter.lookup_expr not in ['exact', 'in']:
+                continue
+
+            # Choose the lookup expression map based on the filter type
+            if isinstance(existing_filter, (
+                MultiValueDateFilter,
+                MultiValueDateTimeFilter,
+                MultiValueNumberFilter,
+                MultiValueTimeFilter
+            )):
+                lookup_map = FILTER_NUMERIC_BASED_LOOKUP_MAP
+
+            elif isinstance(existing_filter, (
+                django_filters.ModelChoiceFilter,
+                django_filters.ModelMultipleChoiceFilter,
+                TreeNodeMultipleChoiceFilter,
+                TagFilter
+            )) or existing_filter.extra.get('choices'):
+                # These filter types support only negation
+                lookup_map = FILTER_NEGATION_LOOKUP_MAP
+
+            elif isinstance(existing_filter, (
+                django_filters.filters.CharFilter,
+                django_filters.MultipleChoiceFilter,
+                MultiValueCharFilter,
+                MultiValueMACAddressFilter
+            )):
+                lookup_map = FILTER_CHAR_BASED_LOOKUP_MAP
+
+            else:
+                # Do not augment any other filter types with more lookup expressions
+                continue
+
+            # Get properties of the existing filter for later use
+            field_name = existing_filter.field_name
+            field = get_model_field(cls._meta.model, field_name)
+
+            # Create new filters for each lookup expression in the map
+            for lookup_name, lookup_expr in lookup_map.items():
+                new_filter_name = '{}__{}'.format(existing_filter_name, lookup_name)
+                if existing_filter.lookup_expr == 'in':
+                    lookup_expr = 'in'  # 'in' lookups must remain to avoid unwanted slicing on certain querysets
```

Comment:
I suppose that is a possibility, however, the only filters we use `in` on are `region` are `id__in` (which we should deprecate anyway). This exists to create negated region filters.

Open to suggestions on how to better refactor it though.

Reply:
How about adding a separate lookup map for MultipleChoiceFilters?

```python
FILTER_MULTICHOICE_LOOKUP_MAP = dict(
    n='in'
)
```

Reply:
This is only a problem with our region filters because we do this to explicitly set the lookup expression to `in`:
```
region_id = TreeNodeMultipleChoiceFilter(
    queryset=Region.objects.all(),
    field_name='region',
    lookup_expr='in',
    label='Region (ID)',
)
region = TreeNodeMultipleChoiceFilter(
    queryset=Region.objects.all(),
    field_name='region',
    lookup_expr='in',
    to_field_name='slug',
    label='Region (slug)',
)
```

I have fixed this by applying your solution to `TreeNodeMultipleChoiceFilter` and move it before the lookup map matching for `MultipleChoiceFilters` which was somewhat hiding this because of subclassing.